controls/bsi_app_4_4.yml

```diff
@@ -313,51 +313,128 @@ controls:
-      only be changed by authorised persons and management services.
+      (4) All the criteria used as labels for a connection SHOULD be secured in such a way that they 
+      can only be changed by authorised persons and management services.
     notes: >-
-      TBD
-    status: pending
-    rules: []
+      In a cluster using a network plugin that supports Kubernetes network policy, network isolation 
+      is controlled entirely by NetworkPolicy objects. In OpenShift, the default plugins (OpenShift SDN,
+      OVN Kubernetes) supports using network policy. Support for NetworkPolicy objects is verified
+      using rules.
+
+      Section 1-3: By default, all pods in a project are accessible from other pods and network endpoints. 
+      To isolate one or more pods in a project, you need to create NetworkPolicy objects in that project
+      to indicate the allowed incoming connections. If a pod is matched by selectors in one or more 
+      NetworkPolicy objects, then the pod will accept only connections that are allowed by at least 
+      one of those NetworkPolicy objects. A pod that is not selected by any NetworkPolicy objects 
+      is fully accessible. 
+      
+      It is useful to create default policies for each application namespace e.g. to deny all ingress
+      traffic by default. The existance of at least one network policy and the automatic creation
+      as part of a namespace template is checked using rules. 
+      
+      The creation of suitable NetworkPolicy objects that satisfy the requirements from sections 1 to 3, 
+      however, needs to be ensured by the application owner.
+      
+      Section 4: It needs to be ensured organizationally, that only required subjects are granted
+      RBAC to change the relevant Kubernetes objects.
+    status: partial
+    rules:
+      # General support of network policies
+      - configure_network_policies
+      # Section 1-2
+      - configure_network_policies_namespaces
+      - project_config_and_template_network_policy
 
   - id: APP.4.4.A19
     title: High Availability of Kubernetes
     levels:
     - elevated
     description: >-
-      A Kubernetes operation SHOULD be set up in such a way that if a site fails, the clusters (and
-      thus the applications in the pods) either continue to run without interruption or can be
+      (1) A Kubernetes operation SHOULD be set up in such a way that if a site fails, the clusters 
+      (and thus the applications in the pods) either continue to run without interruption or can be
       restarted in a short time at another site.
-      Should a restart be required, all the necessary configuration files, images, user data, network
-      connections, and other resources required for operation (including the necessary hardware)
+      (2) Should a restart be required, all the necessary configuration files, images, user data, 

```

Comment:
not sure if we have a rule for this, but this would also require no local node storage used.

Reply:
It feels too fine-grained for me here. This requirement would also be necessary for any pod high availabily, would't it? Do we have a separate control for that?

For me, section 2 focuses on the "cluster restart in other site" scenario, which is largely to be ensured organizationally

Reply:
there is no control for that in APP.4.4 . you are correct, I am to fine-grained here